### PR TITLE
UltraMsg New Actions

### PR DESCRIPTION
**Pull request by vellames-turing**
- [x] Send Audio
- [x] Send Document
- [x] Send Image
- [x] Send Link
- [x] Send Location
- [x] Send Message
- [x] Send Video

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/ultramsg/ultramsg.app.mjs [hunk 1]
```diff
@@ -1,11 +1,91 @@
...
+      const urlSearchParams = new URLSearchParams();
+      urlSearchParams.append("token", this.$auth.token);
+
+      const keys = Object.keys(data);
+      for (let i = 0; i < keys.length; i++) {
+        urlSearchParams.append(keys[i], data[keys[i]]);
+      }
+
+      return urlSearchParams;
+    },
```
Comment (jcortes) [inline components/ultramsg/ultramsg.app.mjs:35]:
```suggestion
    _getAxiosConvertedData(data = {}) {
      return new URLSearchParams({
        token: this.$auth.token,
        ...data,
      });
    },
```

File: components/ultramsg/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "@pipedream/ultramsg",
+    "version": "0.3.3",
+    "description": "Pipedream Ultramsg Components",
+    "main": "ultramsg.app.js",
```
Comment (jcortes) [inline components/ultramsg/package.json:5]:
```suggestion
    "main": "ultramsg.app.mjs",
```

File: components/ultramsg/action/send-message/send-message.mjs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      label: "Body",
+      description: "Message content",
+    },
+    priority: {
+      type: "integer",
+      label: "priority",
+      description: "Priority on internal queue of your message",
+      default: 10,
+      min: 0,
+    },
```
Comment (jcortes) [inline components/ultramsg/action/send-message/send-message.mjs:28]: Seems like you can set an `options` property with this data
```
priority = 0: for High priority like OTP messages.
priority = 5: used with general messages.
priority =10: Non-urgent promotional offers and notifications to your customers.
```

File: components/ultramsg/action/send-location/send-location.mjs [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Location",
+  description: "Send a location to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/location)",
+  key: "ultramsg-send-a-location",
+  version: "0.0.2",
```
Comment (jcortes) [inline components/ultramsg/action/send-location/send-location.mjs:7]:
```suggestion
  version: "0.0.1",
```

File: components/ultramsg/action/send-audio/send-audio.mjs [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send an Audio",
+  description: "Send an audio to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/audio)",
+  key: "ultramsg-send-an-audio",
```
Comment (jcortes) [inline components/ultramsg/action/send-audio/send-audio.mjs:6]:
```suggestion
  key: "ultramsg-send-audio",
```

File: components/ultramsg/action/send-document/send-doucument.mjs [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Document",
+  description: "Send a document to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/document)",
+  key: "ultramsg-send-a-document",
```
Comment (jcortes) [inline components/ultramsg/action/send-document/send-doucument.mjs:6]:
```suggestion
  key: "ultramsg-send-document",
```

File: components/ultramsg/action/send-image/send-image.mjs [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Image",
+  description: "Send a image to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/image)",
+  key: "ultramsg-send-a-image",
```
Comment (jcortes) [inline components/ultramsg/action/send-image/send-image.mjs:6]:
```suggestion
  key: "ultramsg-send-image",
```

File: components/ultramsg/action/send-link/send-link.mjs [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Link",
+  description: "Send a link to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/link)",
+  key: "ultramsg-send-a-link",
```
Comment (jcortes) [inline components/ultramsg/action/send-link/send-link.mjs:6]:
```suggestion
  key: "ultramsg-send-link",
```

File: components/ultramsg/action/send-location/send-location.mjs [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Location",
+  description: "Send a location to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/location)",
+  key: "ultramsg-send-a-location",
```
Comment (jcortes) [inline components/ultramsg/action/send-location/send-location.mjs:6]:
```suggestion
  key: "ultramsg-send-location",
```

File: components/ultramsg/action/send-message/send-message.mjs [hunk 10]
```diff
@@ -0,0 +1,47 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Message",
+  description: "Send a message to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/chat)",
+  key: "ultramsg-send-a-message",
```
Comment (jcortes) [inline components/ultramsg/action/send-message/send-message.mjs:6]:
```suggestion
  key: "ultramsg-send-message",
```

File: components/ultramsg/action/send-video/send-video.mjs [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send a Video",
+  description: "Send a video to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/video)",
+  key: "ultramsg-send-a-video",
```
Comment (jcortes) [inline components/ultramsg/action/send-video/send-video.mjs:6]:
```suggestion
  key: "ultramsg-send-video",
```

**Review by jcortes — CHANGES_REQUESTED** [review]
Hi @vellames-turing it looks good to me. Just a few minor issues and it will be ready for QA. Thanks!

**Comment by vellames-turing** [comment]
@jcortes Suggestions implemented

**Review by js07 — COMMENTED** [review]
The changes since the last review look good. I just made some minor comments.

File: components/ultramsg/ultramsg.app.mjs [hunk 12]
```diff
@@ -1,11 +1,86 @@
...
+
 export default {
   type: "app",
   app: "ultramsg",
-  propDefinitions: {},
+  propDefinitions: {
+    to: {
+      type: "string",
+      label: "To",
+      description: "Phone with international format e.g. `+1408XXXXXXX` , or chatID for contact or group",
```
[the remark this answers is not in the document]
Comment (vellames-turing) [reply components/ultramsg/ultramsg.app.mjs:10]: Oh, I did not notice these endpoints, I will take a look

File: components/ultramsg/action/send-audio/send-audio.mjs [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import ultramsg from "../../ultramsg.app.mjs";
+
+export default {
+  name: "Send an Audio",
+  description: "Send audio file to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/audio)",
```
Comment (js07) [inline components/ultramsg/action/send-audio/send-audio.mjs:5]:
```suggestion
  name: "Send Audio",
  description: "Send an audio file to a specified number. [See the docs here](https://docs.ultramsg.com/api/post/messages/audio)",
```

**Review by js07 — APPROVED** [review]
@vellames-turing Looks good to me! I just made a minor suggestion.

**Review by js07 — APPROVED** [review]

**Comment by dannyroosevelt** [comment]
@vellames-turing feel free to merge!

**Pull request merged**